harness/determined/common/storage/boto3_credential_manager.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from datetime import timedelta
+from typing import Any, Dict, Optional
+
+import boto3
+from boto3 import Session
+from botocore.credentials import (
+    CredentialProvider,
+    Credentials,
+    RefreshableCredentials,
+    SharedCredentialProvider,
+    _local_now,
+)
+from botocore.session import get_session
+
+
+class Boto3Manager:
+    """
+    Wrapper class around boto3 that allows registering of custom credential providers
+    and provides methods for setting the default boto3 session or fetching a new session
+    with the new context
+    """
+
+    def __init__(self) -> None:
+        self.session = get_session()
```

Comment:
What do you think of using a different strategy, where instead of setting the default session to one we've configured, can we just reconfigure the existing default session? That way if two hacks like this ever have to coexist, we've done our part to work with the other one.